### PR TITLE
Put regcode and OBS credentials to CI env vars

### PR DESCRIPTION
**Pull request by Sergeykot**
I did move the OBS credentials and CI Env vars to build and runtime aregs.

to test it locally, one can set those env vars and then (for example for SLES12SP3)

```
docker build --build-arg OBS_USER=$OBS_USER --build-arg OBS_PASSWORD=$OBS_PASSWORD -t connect.12sp3 -f Dockerfile.12sp3 .
docker run -e VALID_REGCODE=$VALID_REGCODE -e EXPIRED_REGCODE=$EXPIRED_REGCODE -e NOT_ACTIVATED_REGCODE=$NOT_ACTIVATED_REGCODE --privileged --rm -t connect.12sp3 ./docker/prophet_tests.sh
```

For SLES15

```
docker build --build-arg OBS_USER=$OBS_USER --build-arg OBS_PASSWORD=$OBS_PASSWORD -t connect.15sp0 -f Dockerfile.15sp0 .
docker run -e VALID_REGCODE=$VALID_REGCODE -e EXPIRED_REGCODE=$EXPIRED_REGCODE -e NOT_ACTIVATED_REGCODE=$NOT_ACTIVATED_REGCODE --privileged --rm -t connect.15sp0 ./docker/prophet_tests.sh
```

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/492911/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image-sle12sp3. (Merged 392749cf82b12bb9d22dc9a8e8db4ba4a1398b42 into 3f809886f5b50898a0cb23586ed3a5af9565a3f4)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/492913/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image-sle12sp3. (Merged 644c84817f03c5c690f3be02361c3da5d5af56fb into 3f809886f5b50898a0cb23586ed3a5af9565a3f4)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493678/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image-sle12sp3. (Merged 644c84817f03c5c690f3be02361c3da5d5af56fb into 3f809886f5b50898a0cb23586ed3a5af9565a3f4)

File: Dockerfile.12sp0 [hunk 1]
```diff
@@ -13,8 +17,9 @@ RUN zypper ar http://download.suse.de/ibs/SUSE:/SLE-12:/GA/standard/ SLE-12-stan
       vim osc ruby2.1-rubygem-gem2rpm hwinfo libx86emu1 zypper-migration-plugin sudo curl && \
     zypper --non-interactive rr SLE-12-standard SLE-12-update-standard SLE-12-ltss
 
-RUN wget http://username:password@gaffer.suse.de:9999/files/.regcode -O ~/.regcode
-RUN wget http://username:password@gaffer.suse.de:9999/files/.oscrc -O ~/.oscrc
+RUN printf "$REGCODE" >> ~/.regcode
+ADD integration/create-oscrc.sh /tmp/connect/integration/create-oscrc.sh
+RUN sh integration/create-oscrc.sh
```
Comment (josegomezr) [inline Dockerfile.12sp0:22]: this exec doesn't exist in the current working dir. Maybe you meant

```suggestion
RUN sh /tmp/connect/integration/create-oscrc.sh
```

:question:

Comment (Sergeykot) [reply Dockerfile.12sp0]: thanks, changes applied!

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493701/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image-sle12sp3. (Merged ac02792474191b38dac375ee578e8fff4cff3a9b into 3f809886f5b50898a0cb23586ed3a5af9565a3f4)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493704/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 89c8f484de51338137163bce9ef80b232ef47344 into 3f809886f5b50898a0cb23586ed3a5af9565a3f4)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493730/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 89c8f484de51338137163bce9ef80b232ef47344 into 3f809886f5b50898a0cb23586ed3a5af9565a3f4)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493965/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 89c8f484de51338137163bce9ef80b232ef47344 into 8f04e4eb06ec200d179194cb70cf022e3e038db2)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493984/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 89c8f484de51338137163bce9ef80b232ef47344 into 8f04e4eb06ec200d179194cb70cf022e3e038db2)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/493994/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 89c8f484de51338137163bce9ef80b232ef47344 into 8f04e4eb06ec200d179194cb70cf022e3e038db2)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/494051/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 89c8f484de51338137163bce9ef80b232ef47344 into 8f04e4eb06ec200d179194cb70cf022e3e038db2)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/494057/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 8f04e4eb06ec200d179194cb70cf022e3e038db2)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/494289/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 1e30711b4d2559f2739130e5592b399c2dfd3530)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/494318/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 1e30711b4d2559f2739130e5592b399c2dfd3530)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/494917/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 1e30711b4d2559f2739130e5592b399c2dfd3530)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/495744/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 1ee02b73158120e8f8b6329eb3b2e32f283d9886)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/495982/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 8e9435b86e387ac93fa633ebdc8b3604c9e88ae6)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/503919/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/504638/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 1bb50ba68c9197b3595d79ea15365f0d126b00af into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/504940/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image-sle12sp3. (Merged 95f0dff8f0d74ce860d645a31fd90586c7e68200 into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/504948/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 33c54cb706c45dd2c017dce71056115a232eff5d into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/505902/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle15sp0. (Merged fd6911e018b73337b254ac9b347c4fd24f04bf2d into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/505910/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle15sp0. (Merged fd6911e018b73337b254ac9b347c4fd24f04bf2d into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/505913/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle15sp0. (Merged fd6911e018b73337b254ac9b347c4fd24f04bf2d into 6342c2b0db69e7925060a76fe3ec3bf02632aea3)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/506018/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle15sp0. (Merged fd6911e018b73337b254ac9b347c4fd24f04bf2d into 01ed58643c2188f564dbba39ab26c4a0655293eb)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/506124/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle15sp0. (Merged acf27040186e1ecf069870279dcd29b4e5ab2df5 into 01ed58643c2188f564dbba39ab26c4a0655293eb)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/506125/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged ff2c2a4651e4f16ba8befdd3e223a52c99427a0f into 01ed58643c2188f564dbba39ab26c4a0655293eb)

File: Dockerfile.15sp3 [hunk 2]
```diff
@@ -16,13 +19,13 @@ RUN zypper ar http://download.suse.de/ibs/SUSE/Products/SLE-Module-Basesystem/15
     zypper --non-interactive install git-core ruby-devel make gcc gcc-c++ build wget dmidecode \
       vim osc hwinfo libx86emu3 zypper-migration-plugin sudo awk curl
 
-RUN wget http://username:password@gaffer.suse.de:9999/files/.regcode -O ~/.regcode
-RUN wget http://username:password@gaffer.suse.de:9999/files/.oscrc -O ~/.oscrc
+ADD integration/create-oscrc.sh /tmp/connect/integration/create-oscrc.sh
```
Comment (josegomezr) [inline Dockerfile.15sp3:22]: Use copy instead of ADD

see [here](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy)

**Review by digitaltom — APPROVED** [review]
Works nice! :+1: 
I fixed `create-oscrc.sh` to work correctly when the password has special characters.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/508089/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 274c6e9f100c8a9d5b3951ca19af46773b7a1985 into 01ed58643c2188f564dbba39ab26c4a0655293eb)

File: Dockerfile.15sp3 (hunk 2 above)
Comment (Sergeykot) [reply Dockerfile.15sp3]: thanks, good point, I did replace it to `COPY` also in other places

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/508362/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged aea05eaa76f59baf78e44e653ac5eed0aeab62db into 01ed58643c2188f564dbba39ab26c4a0655293eb)

**Pull request merged**